cmd/signals.go

```diff
@@ -59,13 +60,16 @@ func handleSignals() {
 	for {
 		select {
 		case err := <-globalHTTPServerErrorCh:
-			logger.LogIf(context.Background(), err)
-			var oerr error
-			if objAPI := newObjectLayerFn(); objAPI != nil {
-				oerr = objAPI.Shutdown(context.Background())
+			if err == http.ErrCertRevoked {
+				globalIsCertRevoked = true
```

Comment:
it is sort of not a good idea to update global values like this IMHO,  it's a racy venture we should simply have a mechanism in HTTP server itself to reject connections and they don't need to be S3 error XML compatible IMHO.

Reply:
> it is sort of not a good idea to update global values like this IMHO, it's a racy venture we should simply have a mechanism in HTTP server itself to reject connections and they don't need to be S3 error XML compatible IMHO.

I needed to do this style because we will not refuse all connections, but only some of them. When a certificate is revoked, we will only reject S3 & Browser connections. Admin & RPC & other should still be working. So this is an information that I need to spread across all the code.

Reply:
Can we perhaps use a safer model here basically an atomic counter perhaps?